apps/meteor/ee/app/livechat-enterprise/server/hooks/afterTakeInquiry.ts

```diff
@@ -1,25 +1,55 @@
+import type { InquiryWithAgentInfo, IOmnichannelRoom } from '@rocket.chat/core-typings';
+import { LivechatVisitors } from '@rocket.chat/models';
+
+import { RoutingManager } from '../../../../../app/livechat/server/lib/RoutingManager';
+import { afterTakeInquiry } from '../../../../../app/livechat/server/lib/hooks';
 import { settings } from '../../../../../app/settings/server';
-import { callbacks } from '../../../../../lib/callbacks';
+import { AutoTransferChatScheduler } from '../lib/AutoTransferChatScheduler';
 import { debouncedDispatchWaitingQueueStatus } from '../lib/Helper';
+import { OmnichannelQueueInactivityMonitor } from '../lib/QueueInactivityMonitor';
 import { cbLogger } from '../lib/logger';
 
-callbacks.add(
-	'livechat.afterTakeInquiry',
-	async ({ inquiry }) => {
-		if (!settings.get('Livechat_waiting_queue')) {
-			return inquiry;
-		}
+afterTakeInquiry.patch(
+	async (
+		originalFn,
+		{ inquiry, room }: { inquiry: InquiryWithAgentInfo; room: IOmnichannelRoom },
+		agent: { agentId: string; username: string },
+	) => {
+		await originalFn({ inquiry, room }, agent);
 
 		if (!inquiry) {
-			return null;
+			return;
+		}
+
+		if (settings.get('Livechat_waiting_queue')) {
+			const { department } = inquiry;
+			void debouncedDispatchWaitingQueueStatus(department);
+			cbLogger.debug({
+				msg: 'Queue status updated',
+				queue: department || 'public',
+			});
 		}
 
-		const { department } = inquiry;
-		debouncedDispatchWaitingQueueStatus(department);
+		if (settings.get<boolean>('Livechat_last_chatted_agent_routing') && agent && RoutingManager.getConfig()?.autoAssignAgent) {
+			const { v: { token } = {} } = inquiry;
+			if (token) {
+				await LivechatVisitors.updateLastAgentByToken(token, { ...agent, ts: new Date() });
+				cbLogger.info({
+					msg: 'Updated last agent of visitor',
+					token,
+					newAgent: { _id: agent.agentId, username: agent.username },
+				});
+			}
+		}
+
+		const autoTransferTimeout = settings.get<number>('Livechat_auto_transfer_chat_timeout');
+		if (autoTransferTimeout && !(room?.autoTransferredAt || room?.autoTransferOngoing)) {
+			await AutoTransferChatScheduler.scheduleRoom(inquiry.rid, autoTransferTimeout);
+		}
 
-		cbLogger.debug(`Statuses for queue ${department || 'Public'} updated successfully`);
-		return inquiry;
+		const maxQueueWaitTime = settings.get<number>('Livechat_max_queue_wait_time');
+		if (!maxQueueWaitTime || maxQueueWaitTime < 0) {
```

Comment:
https://github.com/RocketChat/Rocket.Chat/blob/c209ea761ab859fc9166b2fafb3c36906149a0b3/apps/meteor/ee/app/livechat-enterprise/server/hooks/autoCloseQueued.ts#L6-L9

based on the old code I think this condition is inverted, and if it is indeed inverted since no test failed it seems we don't this feature covered, pls create a task to create a test if that's the case:
```suggestion
		if (maxQueueWaitTime > 0) {
```

side note: it is unclear why `Livechat_max_queue_wait_time` is used to call `stopInquiry` or not